src/notes.c

```diff
@@ -491,6 +497,48 @@ int git_note_create(
 	return error;
 }
 
+int git_note_commit_create(
+	git_oid *notes_commit_out,
+	git_oid *notes_blob_out,
+	git_repository *repo,
+	git_commit *parent,
+	const git_signature *author,
+	const git_signature *committer,
+	const git_oid *oid,
+	const char *note,
+	int allow_note_overwrite)
+{
+	int error;
+	git_tree *tree = NULL;
+	char *target = NULL;
+	git_oid note_commit_oid, note_blob_oid;
+
+	target = git_oid_allocfmt(oid);
```

Comment:
Not sure I follow your first point about accepting an OID. Do you mean why does `note_write` take target as `char *` rather than `git_oid *` ?

For now I'll just switch to static buffer, and `git_oid_tostr`